tensorflow_datasets/core/dataset_builder.py

```diff
@@ -382,13 +382,24 @@ def _log_download_bytes(self):
     # Print is intentional: we want this to always go to stdout so user has
     # information needed to cancel download/preparation if needed.
     # This comes right before the progress bar.
+
+    stat = os.statvfs('/')
```

Comment:
let's move this into py_utils so that dataset builder just calls:

`if dataset_size > py_utils.free_disk_size(self._data_dir):`

Reply:
Thank you. I moved this into py_utils.

But if call function like this `(self._data_dir)` when I run the test `python -m tensorflow_datasets.scripts.download_and_prepare --datasets=mnist`, I got an error `FileNotFoundError: [Errno 2] No such file or directory: '/Users/user1/tensorflow_datasets/mnist/1.0.0'
` because that file not downloaded. It's check before creating file. So I called the root directory.

Reply:
(looks like you need to push the commit; I'm still seeing the old file)

Reply:
And overall we need to think about doing this in a way that's robust to:
* OS: Linux, Windows
* Machines with multiple disks (i.e. we need to use the one that data_dir is on)
* Cloud filesystems (i.e. if the data_dir were set to `gs://...`; in that case we want to skip this check)

Reply:
I import the file name like this `from tensorflow_datasets.core.utils import py_utils` because I've been warned before about that. But I don't remember clearly, is it problem ? Isn't it clean `from tensorflow_datasets.core.utils.py_utils import free_disk_size` ?

Reply:
We always import modules, never individual items.

Reply:
Okay I got it. Thank you